src/steth/Swapper.sol

```diff
@@ -126,30 +136,66 @@ contract Swapper {
         return amountReceived;
     }
 
-    function lidoSwapWethToWstEth(uint256 _wethAmount) external {
+    /**
+     * Swap WETH to wstETH using Lido for ETH to stETH conversion.
+     * @param _wethAmount Amount of WETH to swap.
+     * @return Amount of wstETH received.
+     */
+    function lidoSwapWethToWstEth(uint256 _wethAmount) external returns (uint256) {
         // weth to eth
         weth.withdraw(_wethAmount);
 
-        // stake to lido / eth => stETH
+        // stake to lido / eth => stEth
         stEth.submit{value: _wethAmount}(address(0x00));
 
-        //  stETH to wstEth
+        //  stEth to wstEth
         uint256 stEthBalance = stEth.balanceOf(address(this));
-        ERC20(address(stEth)).safeApprove(address(wstETH), stEthBalance);
+        ERC20(address(stEth)).safeApprove(address(wstEth), stEthBalance);
 
-        wstETH.wrap(stEthBalance);
+        return wstEth.wrap(stEthBalance);
     }
 
+    /**
+     * Swap stETH to WETH on Curve.
+     * @param _stEthAmount Amount of stETH to swap.
+     * @param _wethAmountOutMin Minimum amount of WETH to receive.
+     * @return wethReceived Amount of WETH received.
+     */
     function curveSwapStEthToWeth(uint256 _stEthAmount, uint256 _wethAmountOutMin)
         external
         returns (uint256 wethReceived)
     {
-        // stETH to eth
+        // stEth to eth
         ERC20(address(stEth)).safeApprove(address(curvePool), _stEthAmount);
 
         wethReceived = curvePool.exchange(1, 0, _stEthAmount, _wethAmountOutMin);
 
         // eth to weth
         weth.deposit{value: address(this).balance}();
     }
+
+    /**
+     * Swap WETH to wstETH using Curve for ETH to stETH conversion.
+     * @param _wethAmount Amount of WETH to swap.
+     * @param _wstEthAmountOutMin Minimum amount of wstETH to receive.
+     * @return wstEthReceived Amount of wstETH received.
+     */
+    function curveSwapWethToWstEth(uint256 _wethAmount, uint256 _wstEthAmountOutMin)
+        public
+        returns (uint256 wstEthReceived)
+    {
+        // weth to eth
+        weth.withdraw(_wethAmount);
+
+        // eth to stEth
+        curvePool.exchange{value: _wethAmount}(0, 1, _wethAmount, 0);
```

Comment:
Instead of 0, how about putting _wethAmount in the minimum amount out received here.

Because curve should atleast give equal amount of stEth right, else we are better off just using lido.

Reply:
resolved in person